### PR TITLE
JAMES-2559 Add original branch parameter in merging tool

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

**Comment by aduprat** [comment]
```
[712] Running shell script
+ docker run -a stdout -a stderr -t --name git-publish-james-2559-52875a40-b2e2-43ee-9277-a8c20adaa403 -v /origin git/publish-james-2559-52875a40-b2e2-43ee-9277-a8c20adaa403 2afebd9c47519d6d7abaeca7f27b373321f52039 MERGE-james-2559-52875a40-b2e2-43ee-9277-a8c20adaa403
Usage : 
./merge.sh SHA1 RESULTING_BRANCH
    BRANCH: the original branch used for merge
    SHA1: SHA1 to merge with the branch
    RESULTING_BRANCH : Resulting branch of the merge
```
as expected

File: dockerfiles/merge/merge.sh [hunk 1]
```diff
@@ -3,29 +3,34 @@
 printUsage() {
    echo "Usage : "
    echo "./merge.sh SHA1 RESULTING_BRANCH"
```
Comment (rouazana) [inline dockerfiles/merge/merge.sh]: you missed to modify this line

File: dockerfiles/merge/merge.sh [hunk 2]
```diff
@@ -3,29 +3,34 @@
...
+RESULTING_BRANCH=$3
 
 APACHE_REPO=`git remote show | grep apache || true`
 if [ -z "$APACHE_REPO" ]; then
     git remote add apache https://github.com/apache/james-project.git
 fi 
+# Getting BRANCH from apache repo
 git fetch apache
-git checkout apache/master -b master
+git checkout apache/$BRANCH -B $BRANCH
```
Comment (rouazana) [inline dockerfiles/merge/merge.sh]: why don't you default on master? This will allow to be compatible with current CI, and it seems a sane default, no?

Comment (aduprat) [reply dockerfiles/merge/merge.sh]: I hesitate on that point, will do.

File: dockerfiles/merge/merge.sh [hunk 3]
```diff
@@ -3,29 +3,34 @@
...
    echo "./merge.sh SHA1 RESULTING_BRANCH"
-   echo "    SHA1: SHA1 to merge with master"
+   echo "    BRANCH: the original branch used for merge"
+   echo "    SHA1: SHA1 to merge with the branch"
    echo "    RESULTING_BRANCH : Resulting branch of the merge"
    exit 1
 }
 
-if [ "$#" -ne 2 ]; then
+if [ "$#" -ne 3 ]; then
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply dockerfiles/merge/merge.sh]: Cqn't we make the 3rd argument optional and, if absent assume `apache/master` ?

File: dockerfiles/merge/merge.sh [hunk 4]
```diff
@@ -2,30 +2,39 @@
...
+fi
 
 APACHE_REPO=`git remote show | grep apache || true`
 if [ -z "$APACHE_REPO" ]; then
     git remote add apache https://github.com/apache/james-project.git
 fi 
+# Getting original branch from apache repo
 git fetch apache
-git checkout apache/master -b master
+git checkout apache/$ORIGINAL_BRANCH -B $ORIGINAL_BRANCH
```
Comment (chibenwa) [inline dockerfiles/merge/merge.sh]: I would have put the repo as part of the origin branch.
It would allow to have branches specific to Linagora.

**Comment by chibenwa** [comment]
Merging step passed

File: dockerfiles/merge/merge.sh (hunk 4 above)
Comment (aduprat) [reply dockerfiles/merge/merge.sh]: That's another story ;)

**Comment by chibenwa** [comment]
Merged

File: dockerfiles/merge/merge.sh (hunk 4 above)
Comment (rouazana) [reply dockerfiles/merge/merge.sh]: why? what does it change to do it know?

I find it a shame that we need to change the api just after...

**Pull request closed without merging**